crates/iota/src/upgrade_compatibility.rs

```diff
@@ -413,6 +423,57 @@ impl CompatibilityMode for CliCompatibilityMode {
     }
 }
 
+const COMPATIBILITY_PREFIX: &str = "Compatibility ";
+
+/// Generates an enum Category along with individual enum for each individual
+/// category and impls conversions for each category.
+macro_rules! upgrade_codes {
+    ($($cat:ident: [
+        $($code:ident: { msg: $code_msg:literal }),* $(,)?
+    ]),* $(,)?) => {
+        #[derive(PartialEq, Eq, Clone, Copy, Debug, Hash, PartialOrd, Ord)]
+        #[repr(u8)]
+        pub enum Category {
+            #[allow(dead_code)]
+            ZeroPlaceholder,
+            $($cat,)*
+        }
+
+        $(
+            #[derive(PartialEq, Eq, Clone, Copy, Debug, Hash)]
+            #[repr(u8)]
+            pub enum $cat {
+                #[allow(dead_code)]
+                ZeroPlaceholder,
+                $($code,)*
+            }
+
+            impl From<$cat> for DiagnosticInfo {
```

Comment:
Do you mind reverting this to upstream please? Take a note to do changes later but it would make our lives simpler to be 1:1 for these

Reply:
We've made other changes to the other PRs, so why is this different? I doubt this will make things any more difficult regardless, since they've only made one additional change here and it's to add `#[allow(clippy::from_over_into)]` 😆

Reply:
fine, done